clang/lib/Parse/ParseExpr.cpp

```diff
@@ -4027,14 +4107,15 @@ bool Parser::ConsumeAndStoreBoundsExpression(CachedTokens &Toks) {
 
 /// Given a list of tokens that have the same shape as a bounds
 /// expression, parse them to create a bounds expression.  Delete
-/// the list of tokens at the end.
-///
+/// the list of tokens at the end. Also parses where clause occurring on a
+/// decl.
```

Comment:
The last line of this comment looks like an after-thought. Could you please integrate it more with the body of the entire comment? Also, in the comment lines below, perhaps you could mention that the resulting `where` clause is attached to `ThisDecl`.